packages/plugin-fsapp/src/index.ts

```diff
@@ -0,0 +1,452 @@
+/// <reference types="@brandingbrand/code-cli-kit/types" />
+
+/**
+ * Defines a plugin for @brandingbrand/code-cli-kit.
+ * @module EnvProcessingPlugin
+ */
+
+import semver from 'semver';
+import {glob} from 'glob';
+import {types as t} from '@babel/core';
+import {parse} from '@babel/parser';
+import traverse from '@babel/traverse';
+import {
+  definePlugin,
+  fs,
+  logger,
+  path,
+  type BuildConfig,
+  type PrebuildOptions,
+  withPbxproj,
+  getReactNativeVersion,
+  withUTF8,
+  string,
+} from '@brandingbrand/code-cli-kit';
+import chalk from 'chalk';
+import dedent from 'dedent';
+
+/**
+ * Determines if a string represents a package or a file path.
+ *
+ * @param {string} str - The string to check.
+ * @return {boolean} True if the string represents a package, false if it represents a file path.
+ */
+export function isPackage(str: string): boolean {
+  // Regular expression pattern to match package names
+  const packagePattern =
+    /^(@[a-z0-9-~][a-z0-9-._~]*\/)?[a-z0-9-~][a-z0-9-._~]*$/;
+
+  // Check if the string matches the package pattern
+  return packagePattern.test(str);
+}
+
+/**
+ * Asynchronously bundles and requires a package or file.
+ *
+ * @param {string} packageOrFilePath - The name of the package or the file path to bundle and require.
+ * @returns {Promise<any>} A Promise that resolves to the required module.
+ */
+export async function bundleRequire(
+  packageOrFilePath: string,
+  format: 'cjs' | 'esm' = 'cjs',
+): Promise<any> {
+  // Import the 'bundle-require' esm module dynamically
+  // Due to esm and exports + types need to ignore ts for this import
+  // @ts-ignore
+  const {bundleRequire: _bundleRequire} = await import('bundle-require');
+
+  // Check if the input string represents a package
+  if (isPackage(packageOrFilePath)) {
+    // Parse package.json contents
+    const pkg = require(require.resolve(`${packageOrFilePath}/package.json`));
+
+    // Determine if the package is esm, if so reassign format
+    if (pkg.type === 'module') {
+      format = 'esm';
+    }
+
+    // Resolve the package name to its filepath
+    packageOrFilePath = require.resolve(packageOrFilePath, {
+      paths: [process.cwd()],
+    });
+  }
+
+  // Use 'bundle-require' to bundle and require the specified file or package
+  const {mod} = await _bundleRequire({
+    filepath: packageOrFilePath,
+    format,
+  });
+
+  // Return the required module
+  return mod;
+}
+
+async function validateEnvFiles(baseDir: string): Promise<string[]> {
+  const pattern = path.join(baseDir, '**', 'env.*.ts');
+  const files = glob.sync(pattern, {ignore: '**/node_modules/**'});
```

Comment:
We can collect files here but how can we guarantee that these are intended to be env files...we shouldn't just throw an error due to it being named in a particular way but not conforming to a format. this could be a different file used for something else. something to think about